### DISCUSSION TITLE
How to `brew audit` a Formula locally?

### DISCUSSION THREAD

**Discussion by bartekpacia** [Homebrew/discussions#4243] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.1-91-g6b7ecd1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6b7ecd18e9b27e0a42a66ba8a955e7b57d06af50
Last commit: 3 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 34b732b38991b8b2f26b592d87287ebc77937f19
Core tap last commit: 12 days ago
Core tap branch: master
Core tap JSON: 20 Feb 13:02 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BAT_CONFIG_PATH: /Users/bartek/.bat.conf
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: micro
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ENV_HINTS: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.2 => /opt/homebrew/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  szip

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  libftdi0
  libhid

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python@3.11
  szip
```

### Description of issue

I have a problem with `brew audit` command. I cannot get it to work the way I want.

My use case is that I want to run `brew audit` on GitHub Actions when a new Pull Request is made to the repository containing Formulas.

I'll post my whole Actions, since it's not that big and should be easy to understand even for people not using GitHub Actions:

```yaml
name: prepare

on: pull_request

jobs:
  main:
    runs-on: macos-latest

    steps:
      - name: Clone repository
        uses: actions/checkout@v3

      - name: brew tap
        run: brew tap leancodepl/tools

      - name: brew install
        run: cat ./Formula/mobile-tools.rb && brew install ./Formula/mobile-tools.rb

      - name: brew audit
        run: brew audit mobile-tools --strict --online
```

The problem is that instead of using a local formula (the one that is modified in a PR), it checks the currently published version, which doesn't make sense in this situation.

[Here's a PR](https://github.com/leancodepl/homebrew-tools/pull/6) in which I'm trying to implement this Action.

I'd appreciate any help :)

**Comment by Bo98** [comment]
You clone the repository to `GITHUB_WORKSPACE` (Homebrew doesn't read from there) in the first step. You then `brew tap` which will install a tap to the right place, but it will download the state of the repo on `master` - not your PR.

You can replace the first two steps with the `Homebrew/actions/setup-homebrew@master` action which does all the setup work for you.

**Reply by bartekpacia** [reply]
Thanks for you answer :)

> but it will download the state of the repo on master - not your PR

Yeah, that's the problem I have now. How can I download state of the repo from my PR?

**Reply by carlocab** [reply]
Use the [`setup-homebrew`](https://github.com/Homebrew/actions/tree/master/setup-homebrew) action that Bo mentioned above. Or you can look at the relevant code behind the `setup-homebrew` action and replicate it in your workflow.

**Reply by bartekpacia** [reply]
Thanks for helping me @carlocab. I added the mentioned action, but now it seems that `brew audit` doesn't actually audits? [I purposefully added code that `brew audit` warns about ](https://github.com/leancodepl/homebrew-tools/pull/6/commits/8e9797d5cd4e94445547c97e383548fa294c0664)("use Ruby 1.9 hash syntax"), but [`brew audit` is not warning about it](https://github.com/leancodepl/homebrew-tools/actions/runs/4224473096/jobs/7335469210).

**Reply by bartekpacia** [reply]
For future readers, here's how I achieved it: [one](https://github.com/leancodepl/homebrew-tools/blob/e664c9f3245489bfcfe5f1d6c01ac534c39e568f/.github/workflows/prepare.yaml) and [two](https://github.com/leancodepl/homebrew-tools/blob/e664c9f3245489bfcfe5f1d6c01ac534c39e568f/brew_test) and [three](https://github.com/leancodepl/homebrew-tools/blob/e664c9f3245489bfcfe5f1d6c01ac534c39e568f/brew_audit)
